nntrainer/utils/node_exporter.cpp

```diff
@@ -81,23 +118,194 @@ void Exporter::saveTflResult(const std::tuple<props::Unit> &props,
     std::vector<Tensor> new_weights;
     new_weights.reserve(weights.size());
 
-    // std::cerr << "weights! " << weights.size() << ' ' <<
-    // new_weights.capacity() << std::endl; std::transform(weights.begin(),
-    // weights.end(),
-    //                std::back_inserter(new_weights),
-    //                [](const Tensor *t) { return t->clone(); });
-    // std::cerr << "22\n";
     new_weights.push_back(weights[0]->transpose("0:2:1"));
     new_weights.push_back(*weights[1]);
-    // std::cerr << "33\n";
     return new_weights;
   };
   tf_node->setWeightTransformFn(weight_transform);
 
   tf_node->setOpType(tflite::BuiltinOperator_FULLY_CONNECTED);
-  /// we probably going to need flatbuffer inside exporter regarding this
+  auto options = tflite::CreateFullyConnectedOptions(*fbb).Union();
   tf_node->setBuiltinOptions(tflite::BuiltinOptions_FullyConnectedOptions,
+                             options);
+}
+
+template <>
+void Exporter::saveTflResult(const std::tuple<props::Activation> &props,
+                             const ActivationLayer *self) {
+  createIfNull(tf_node);
+
+  auto activation = std::get<props::Activation>(props);
+  switch (activation.get()) {
+  case ActivationType::ACT_RELU: {
+    tf_node->setOpType(tflite::BuiltinOperator_RELU);
+    tf_node->setBuiltinOptions(tflite::BuiltinOptions_NONE,
+                               flatbuffers::Offset<void>() /** no options **/);
+    break;
+  }
+  case ActivationType::ACT_SOFTMAX: {
+    tf_node->setOpType(tflite::BuiltinOperator_SOFTMAX);
+    auto optinos = tflite::CreateSoftmaxOptions(*fbb, 1.0).Union();
+    tf_node->setBuiltinOptions(tflite::BuiltinOptions_SoftmaxOptions, optinos);
+    break;
+  }
+  default:
+    throw std::runtime_error{"Unsupported activation type"};
+  }
+}
+
+template <>
+void Exporter::saveTflResult(
+  const std::tuple<props::FilterSize, std::array<props::KernelSize, CONV2D_DIM>,
+                   std::array<props::Stride, CONV2D_DIM>, props::Padding2D>
+    &props,
+  const Conv2DLayer *self) {
+  createIfNull(tf_node);
+
+  auto weight_transform = [](std::vector<const Tensor *> &old_weights) {
+    std::vector<Tensor> new_weights;
+
+    auto &filter_weight = *old_weights[0];
+    Tensor filter(filter_weight);
+    // tflite filter has shape format {channel_out, height, width, channel_in}
+    convertNCHWToNHWC(filter_weight, filter);
+    new_weights.push_back(filter);
+
+    auto &bias_weight = *old_weights[1];
+    TensorDim bias_dim{std::bitset<4>(0b0001)};
+    bias_dim.setTensorDim(
+      3 /** index **/,
+      bias_weight
+        .channel() /** value **/); // effective dimension = {bias->channel()}
+    Tensor bias(bias_dim);
+    bias.setName(bias_weight.getName());
+    bias.setValue(bias_weight.getValue(
```

Comment:
I guess this makes the bias zero. but bias also need to be trained according to the options. So, I guess we shouldn't set the value as zero.